core/modules/wdb/QueryAction.cc

```diff
@@ -311,29 +314,33 @@ bool QueryAction::Impl::_dispatchChannel() {
     }
     ChunkResourceRequest req(_chunkResourceMgr, m);
 
-    for(int i=0; i < m.fragment_size(); ++i) {
-        wbase::Task::Fragment const& f(m.fragment(i));
-        ChunkResource cr(req.getResourceFragment(i));
-        // Use query fragment as-is, funnel results.
-        for(int qi=0, qe=f.query_size(); qi != qe; ++qi) {
-            MYSQL_RES* res = _primeResult(f.query(qi));
-            if(!res) {
-                erred = true;
-                continue;
-            }
-            if(firstResult) {
-                _fillSchema(res);
-                firstResult = false;
-                numFields = mysql_num_fields(res);
-            } // TODO: may want to confirm (cheaply) that
-            // successive queries have the same result schema.
-            // Now get rows...
-            if(!_fillRows(res, numFields)) {
-                erred = true;
-            }
-            _mysqlConn->freeResult();
-        } // Each query in a fragment
-    } // Each fragment in a msg.
+    try {
+        for(int i=0; i < m.fragment_size(); ++i) {
+            wbase::Task::Fragment const& f(m.fragment(i));
+            ChunkResource cr(req.getResourceFragment(i));
+            // Use query fragment as-is, funnel results.
+            for(int qi=0, qe=f.query_size(); qi != qe; ++qi) {
+                MYSQL_RES* res = _primeResult(f.query(qi));
+                if(!res) {
+                    erred = true;
+                    continue;
+                }
+                if(firstResult) {
+                    _fillSchema(res);
+                    firstResult = false;
+                    numFields = mysql_num_fields(res);
+                } // TODO: may want to confirm (cheaply) that
+                // successive queries have the same result schema.
+                // Now get rows...
+                if(!_fillRows(res, numFields)) {
+                    erred = true;
+                }
+                _mysqlConn->freeResult();
+            } // Each query in a fragment
+        } // Each fragment in a msg.
+    } catch(sql::SqlErrorObject const& e) {
```

Comment:
Good enough for now, but could you add a "TODO fritzm" here to revisit/check this error handling strategy?  It's not clear to me that we want to throw all out of this loop on an error (rather than continuing on and reporting on all errors in the disptach via the multi-error?)